docs/resolver-template/cmd/demoresolver/main.go

```diff
@@ -91,3 +91,8 @@ func (*myResolvedResource) Data() []byte {
 func (*myResolvedResource) Annotations() map[string]string {
 	return nil
 }
+
+// Source returns the source references of the data. None atm.
+func (*myResolvedResource) Source() *v1alpha1.ConfigSource {
+	return nil
+}
```

Comment:
Hey @abayer 

> RE: "So there's the additional problem that we can't guarantee structured resolution source information - that's going to depend on the resolver implementation itself. We can control that for git, bundles, cluster, and hub, but not for any third-party resolvers that are written." from https://github.com/tektoncd/pipeline/issues/5529#issuecomment-1252845275

I feel like the individual resolver implementation just needs to implement the `Source()` method to return a configsource, which is part of `framework.ResolvedResource` interface. This resolver template is an example. That said, I think we just need to update [this section](https://github.com/tektoncd/pipeline/blob/main/docs/how-to-write-a-resolver.md#the-resolve-method) in how-to-write-a-resolver to mention this `Source()` method other than `Data()` and `Annotations()`.

Thoughts on this? 
Thank you!

Reply:
Updated `how-to-write-a-resolver` doc in this PR. Please take a look.